### ISSUE TITLE
Can't access data variables in a forEach loop

### ISSUE THREAD

**Issue by ShadeAJ1** [vuejs/vue#11261]
### Version
2.6.11

### Reproduction link
[https://jsfiddle.net/ndz8xraq/1/](https://jsfiddle.net/ndz8xraq/1/)

### Steps to reproduce
Make a data field as follows
```
data() {
    return {
        names: [ ]
    }
}
```
Here's a method with what kind of logic I am trying to implement.

```
method() {
    getProfiles: function() {
        profiles = queryData
        profiles.forEach(profile => {
            this.names.push(profile.name)
        });
    }
} ,
created() {
    this.getProfiles()
}
````

### What is expected?
What is expected is that for each profile in the profiles list queries from a database, the profiles name should be pushed to the array of names. 

### What is actually happening?
However, when writing this code, you would get names is not defined, even though it is defined in `data()`. 'this.names` value would not change and be set to or pushed to by the forEach loop.

---
To sum up, I cannot find a way to set variables outside of the forEach loop to a value inside the forEach loop. I even tried using local variables and setting the names variable to the local var, but to no avail. On a side note, I basically do the query and have a '.then()' in which I use the forEach in. I would appreciate any help I can get. Please let me know and have a good day!

**Comment by jacekkarczmarczyk** [comment]
You're reproduction is full of errors
https://jsfiddle.net/y7u4gt5a/

**Comment by Justineo** [comment]
Hi, thanks for your interest but Github issues are for bug reports and feature requests only. You can ask questions on the [forum](http://forum.vuejs.org/), the [Discord server](https://vue-land.js.org/), [gitter](https://gitter.im/vuejs/vue) or [StackOverflow](http://stackoverflow.com/tags/vue.js).
